README.md

```diff
@@ -10,4 +10,400 @@ TODO: description
+- `password` _{string}_ - The new password (optional)
+
+**Returns**
+
+- A success message
+
+**Throws**
+
+- `401` if the user is not logged in
+- `400` if `username` or `password` is not in correct format
+- `400` if `username` and `password` are both missing
+
+#### `DELETE /api/accounts/credentials/:username` - Delete a credential from the shared account
+
+**Returns**
+
+- A success message
+
+**Throws**
+
+- `401` if the user is not logged in
+- `404` if no credential with username `username` exists
+- `403` if the credential `username` doesn't belong to the user's account
+
+#### `POST /api/users/session` - Sign in user
+
+**Body**
+
+- `username` _{string}_ - The user's username
+- `password` _{string}_ - The user's password
+
+**Returns**
+
+- A success message
+- The shared account's details
+
+**Throws**
+
+- `403` if the user is already logged in
+- `400` if username or password is not in correct format or missing
+- `401` if the user login credentials are invalid
+
+#### `DELETE /api/users/session` - Sign out user
+
+**Returns**
+
+- A success message
+
+**Throws**
+
+- `401` if user is not logged in
+
+### Insurance Card
+
+#### `GET /api/insurance-cards` - Get a user's insurance cards
+
+**Returns**
+
+- An array of insurance cards created by the user in alphabetical order with respect to their purpose
+
+**Throws**
+
+- `401` if user is not logged in 
+
+#### `POST /api/insurance-cards` - Create a new insurance card
+
+**Body**
+
+- `subscriber_name` _{string}_ - The insurance card's subscriber 
+- `member_id` _{string}_ - The insurance card's member ID
+- `group_number` _{string}_ - The insurance card's group number
+- `plan_number` _{string}_ - The insurance card's plan number
+- `plan_type` _{string}_ - The insurance card's plan type 
+- `purpose` _{string}_ - The insurance card's purpose 
+- `notes` _{string}_ - Notes about the insurance card 
+
+**Returns**
+
+- A success message
+- A object with the created insurance card
+
+**Throws**
+
+- `401` if the user is not logged in
+- `400` If the subscriber_name or purpose is empty or a stream of empty spaces
+
+#### `PATCH /api/insurance-cards/:insuranceCardId?` - Edit the information of an insurance card
+
+**Body**
+
+- `subscriber_name` _{string}_ - The insurance card's subscriber (optional)
+- `member_id` _{string}_ - The insurance card's member ID (optional)
+- `group_number` _{string}_ - The insurance card's group number (optional)
+- `plan_number` _{string}_ - The insurance card's plan number (optional)
+- `plan_type` _{string}_ - The insurance card's plan type (optional)
+- `purpose` _{string}_ - The insurance card's purpose (optional)
+- `notes` _{string}_ - Notes about the insurance card (optional)
+
+**Returns**
+
+- A success message
+- A object with the edited insurance card
+
+**Throws**
+
+- `401` if the user is not logged in
+- `403` if the user is not the owner of the insurance card
+- `400` If the subscriber_name or purpose is empty or a stream of empty spaces
```

Comment:
If these fields are optional and empty, should it be assume that they're unchanged in this edit?